frontend/test/metabase/scenarios/onboarding/setup/user_settings.cy.spec.js

```diff
@@ -261,3 +213,18 @@ describe("user > settings", () => {
     });
   });
 });
+
+/**
+ * Stub the current user authentication method
+ *
+ * @param {Object} authenticationMethod
+ */
+function stubCurrentUser(authenticationMethod) {
+  cy.request("GET", "/api/user/current").then(({ body: user }) => {
+    cy.intercept(
+      "GET",
+      "/api/user/current",
+      Object.assign({}, user, authenticationMethod),
+    ).as("getUser");
+  });
```

Comment:
What do you think about using [routeHandler](https://docs.cypress.io/api/commands/intercept#Controlling-the-response) to modify response data directly, instead of calling the actual API and then spy it?

```suggestion
  cy.intercept("GET", "/api/user/current", req => {
    req.reply(response => {
      response.body = Object.assign({}, response.body, authenticationMethod);
    });
  }).as("getUser");
```

passing a function to `req.reply` is undocumented, but I found it from https://timdeschryver.dev/blog/how-to-patch-an-http-response-with-cypress

Another way which is documented, but feel a bit unpleasant is
```suggestion
  cy.intercept("GET", "/api/user/current", req => {
    req.on("response", res => {
      res.send(Object.assign({}, res.body, authenticationMethod));
    });
  }).as("getUser");
```

which feels like it needs to listen to `response` event and then call `res.send` adding an extra step when compared to using `req.reply` and modify the response directly.

I think this is more idiomatic since we're not calling the API beforehand then stub, but modify the response on the fly in one go.